tests/unit/test_charm.py

```diff
@@ -182,3 +182,11 @@ def test_charm_blocks_if_specified_port_invalid(self):
 
         self.assertIsInstance(self.harness.model.unit.status, BlockedStatus)
         self.assertEqual({}, dict(relation_data))
+
+    def test_non_leader_unit_sets_waiting_status(self):
+        """Test units that are not leader are marked inactive."""
+        self.harness.set_leader(False)
+
+        self.harness.update_config({"targets": "https://192.186.1.0:1234"})
+
+        self.assertEqual(self.harness.model.unit.status, WaitingStatus("inactive unit"))
```

Comment:
Exact comparison to a literal seems unnecessarily fragile.

```suggestion
        self.assertIsInstance(self.harness.model.unit.status, WaitingStatus)
```

Reply:
In this case I do want to check that the message is indeed "inactive unit" because that has been the requirement.